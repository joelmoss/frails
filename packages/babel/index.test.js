const serializer = require('jest-serializer-path')
const { createConfig } = require('@frails/core')

const babel = require('.')
jest.mock('./require_resolve')

describe('babel', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('createConfig', () => {
    expect(createConfig(babel)).toMatchSnapshot()
    expect(JSON.stringify(createConfig(babel()))).toBe(JSON.stringify(createConfig(babel)))
  })

  test('disabled', () => {
    expect(createConfig(babel(false))).toMatchSnapshot()
  })

  test('with options', () => {
    expect(createConfig(babel({ cacheCompression: false }))).toMatchSnapshot()
  })
})
