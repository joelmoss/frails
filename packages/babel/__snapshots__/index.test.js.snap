// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel createConfig 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.js\\$/i,
        "use": Object {
          "loader": "babel-loader",
          "options": Object {
            "cacheDirectory": true,
          },
        },
      },
    ],
  },
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`babel disabled 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`babel with options 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.js\\$/i,
        "use": Object {
          "loader": "babel-loader",
          "options": Object {
            "cacheCompression": false,
            "cacheDirectory": true,
          },
        },
      },
    ],
  },
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;
